src/corecel/sys/KernelParamCalculator.device.hh

```diff
@@ -214,5 +218,29 @@ auto KernelParamCalculator::operator()(size_type min_num_threads) const
     return result;
 }
 
+//---------------------------------------------------------------------------//
+/*!
+ * Maximize L1 cache size if the kernel doesn't use static shared mem.
+ */
+template<class F>
+void KernelParamCalculator::set_carveout(F* kernel_func_ptr) const
+{
+#ifdef CELER_DEVICE_SOURCE
+    CELER_DEVICE_PREFIX(FuncAttributes) attr;
+    CELER_DEVICE_CALL_PREFIX(FuncGetAttributes(
+        &attr, reinterpret_cast<void const*>(kernel_func_ptr)));
```

Comment:
Since the getattr call is so expensive on some platforms (~1s on some AMD machines) it would be better if we could reuse the attributes we already fetched (might need to refactor the kernel registry return value).